syntax = "proto3";
package monitor;

import "google/protobuf/empty.proto";
import "Net_Info.proto";
import "Mem_Info.proto";
import "Cpu_Stat.proto";
import "Cpu_Softirq.proto";
import "Cpu_Load.proto";

message MonitorInfo
{
    string name = 1;
    repeated SoftIrq sort_irq = 2;
    CpuLoad cpu_load = 3;
    repeated CpuStat cpu_stat = 4;
    MemInfo mem_info = 5;
    repeated NetInfo net_info = 6;
}

service GrpcManager
{
    rpc SetMonitorInfo(MonitorInfo) returns (google.protobuf.Empty)
    {}

    rpc GetMonitorInfo(google.protobuf.Empty) returns (MonitorInfo)
    {}
}